fix: Upgrade to Business Edition - Docker (#1238)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/upgrade-to-business-edition/docker.mdx

---
description: Follow the steps to upgrade the Appsmith community Docker installation to a Business Edition.
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Docker

The page provides steps to upgrade the Appsmith Docker Community Edition to the Business Edition.

## Prerequisites

- Sign up on [customer.appsmith.com](https://customer.appsmith.com/) and generate a trial license key.
- Ensure you have at least 2 GB of free storage space for running the backup process.
- Create a backup of the Appsmith instance before upgrading. For more information, see [Backup Appsmith Instance](/getting-started/setup/instance-management/appsmithctl#backup-instance)

> - Ensure that you have the latest Appsmith version deployed on your instance. To update Appsmith, see the [Update Appsmith](/getting-started/setup/instance-management/update-appsmith) section.

Reviewer: ```suggestion
- Ensure that you have the latest Appsmith version deployed on your instance. To update Appsmith version, see [Update Appsmith](/getting-started/setup/instance-management/update-appsmith).
```

> ## Upgrade Appsmith instance

Reviewer: ```suggestion
## Upgrade to Business Edition
```

Follow these steps to upgrade your Appsmith instance:

> 1. Go to the directory where the `docker-compose.yml` file is located.

Reviewer: Do we really need this step? the step below already says Open the `docker-compose.yml` file. Do we need to explicitly ask them to go there. If there is a path/directory we want to instruct the users to go, then this step would make sense.

Author: Hmm. Let me rewrite this.

2. Open the `docker-compose.yml` file, and search for the `image:` key.
3. Update the value of the `image:` key as below:

``` yaml
  services:
    appsmith:
    #highlight-next-line
      image: index.docker.io/appsmith/appsmith-ee
```

> 4. Save the file

Reviewer: ```suggestion
4. Save the file.
```

5. Recreate the instance with:

```bash

> docker-compose up -d --force-recreate

Reviewer: This would work for the replace license situation as well, right?

Author: The license key is no longer needed to be updated in the `docker-compose.yml` file, so the above command should suffice. CC: @sum35h

```

6. Open [https://localhost](https://localhost) and wait for the server to come up. This can take up to 5 minutes. Once the server is up and running, verify the upgrade with:

```bash
# verify the image name as (appsmith-ee) in use 
docker ps
```
You have successfully upgraded your Docker instance to the Business Edition. 

> 7.  Log into your Appsmith account, and enter your license key to activate the instance.

Reviewer: nitpick. too many your look odd

## Troubleshooting

> If you come across any errors, begin by [Restoring the Appsmith instance](/getting-started/setup/instance-management/appsmithctl?current-command-type=docker-commands#restore-instance) to the [backup version](#prerequisites) and contact [support@appsmith.com](mailto:support@appsmith.com).

Reviewer: ```suggestion
If you experience issues during the upgrade, [restore the Appsmith instance](/getting-started/setup/instance-management/appsmithctl?current-command-type=docker-commands#restore-instance) to the backup created in the [prerequisites](#prerequisites) and contact the support team at [support@appsmith.com](mailto:support@appsmith.com).
```
